### PR TITLE
Adjust layout for consistency when building visualizations

### PR DESCRIPTION
**Pull request by jernestmyers**
Addresses #924

The layout takes into account these [notes from the 04/11/2022 UX meeting](https://github.com/VEuPathDB/web-eda/issues/924#issuecomment-1095605698). The build progression looks as follows on a `1280x900` resolution, simulated via Firefox's dev tools:

1. New visualization with no selections. Of note, the `VariableCoverageTable` is not yet rendered.
image
2. Select x and y variables
image
3. Select an overlay
image
4. Select a facet
image

### CODE REVIEW

File: src/lib/core/components/VariableCoverageTable.tsx [hunk 1]
```diff
@@ -59,7 +59,10 @@ export function VariableCoverageTable({
       ? 'VariableCoverageTable'
       : `${containerClassName} VariableCoverageTable`;
 
-  return (
+  return variableSpecs[0].display ||
+    variableSpecs[1].display ||
+    variableSpecs[2].display ||
+    variableSpecs[3].display ? (
```
Comment (bobular) [inline src/lib/core/components/VariableCoverageTable.tsx:65]: I was getting a runtime error about variableSpecs[3] being undefined.  One solution is below

```suggestion
  return variableSpecs[0]?.display ||
    variableSpecs[1]?.display ||
    variableSpecs[2]?.display ||
    variableSpecs[3]?.display ? (
```

Another might be something like `variableSpecs.some( (spec) => spec.display != null )` (if the test for not null is appropriate)

**Review by bobular — COMMENTED** [review]
Just letting you have this comment for now. Will continue more testing

File: src/lib/core/components/VariableCoverageTable.tsx (hunk 1 above)
Comment (bobular) [reply src/lib/core/components/VariableCoverageTable.tsx:65]: I have continued testing with the `some(...)` option and that seems good.

Could you talk with @moontrip if he's available and talk about widening the legend to the standard 400px width that the variable coverage table and birds eye are using?

Comment (jernestmyers) [reply src/lib/core/components/VariableCoverageTable.tsx:65]: I noticed with this approach that histograms are assigned `Main` as the `spec.display` and that bar plots are not. This resulted in the table rendering for histograms before selecting any variables.

I'm changing to `variableSpecs.some( (spec) => spec.variable)` to account for this.

**Review by dmfalke — COMMENTED** [review]
The code looks good. I will look at the UI shortly.

**Comment by dmfalke** [comment]
The UI looks pretty good, but I would suggest making the legend use the width available from its container, and remove the top/bottom margin from the right side.

**Before**
image

**After**
image

**Comment by jernestmyers** [comment]
> Could you talk with @moontrip if he's available and talk about widening the legend to the standard 400px width that the variable coverage table and birds eye are using?

We were having this discussion earlier with @bobular and he spoke with @danicahelb about it. She suggested we not deal with the legend width until b58. Is the final verdict to wait until b58 or to sneak it in now?

**Comment by bobular** [comment]
We could sneak it in if it's not likely to break anything.  The 400px wide legend with labels cut off less than half-way across will look sillier than the narrow legend IMO.  I'm just looking to see where it's done.  I don't think web-component's PlotlyPlot.tsx handles this any more.

**Comment by dmfalke** [comment]
I think we should wait. It's a minor bother to me, but it sounds like there are a host of other related issues that might arise.

**Comment by bobular** [comment]
I had a look at web-components' PlotLegend.tsx and I don't think we should rush this.
Ideally that component should have a width prop, from which the ellipsis threshold is calculated (I know this is not an exact science).  So let's punt on this now and save it for https://github.com/VEuPathDB/web-components/issues/344 in 58

**Review by bobular — APPROVED** [review]

**Review by dmfalke — APPROVED** [review]

**Comment by dmfalke** [comment]
> I had a look at web-components' PlotLegend.tsx and I don't think we should rush this. Ideally that component should have a width prop, from which the ellipsis threshold is calculated (I know this is not an exact science). So let's punt on this now and save it for [VEuPathDB/web-components#344](https://github.com/VEuPathDB/web-components/issues/344) in 58

Now that this is being rendered as HTML, rather than SVG, we should be able to use the CSS `text-overflow: ellipsis` rule to automatically truncate based on available space, rather than truncating the text in JavaScript. This will allow things to be more responsive. I will update the issue linked above.

**Pull request merged**